refactor(ivy): cleanup tests from render3

cexbrayat opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?

`render3` contains tests for JS blocks that are no longer planned to be supported.


## What is the new behavior?

The `di_spec.ts` tests were already migrated to acceptance with `ngFor`/`ngIf`, but were leftover in case JS blocks ended up supported in Ivy. They are now removed.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


## Other information

cexbrayat committed:
refactor(ivy): cleanup di tests from render3

The tests were already migrated to acceptance with `ngFor`/`ngIf`, but were leftover in case JS blocks ended up supported in Ivy.

kara commented on packages/core/test/render3/control_flow_spec.ts:
```diff
@@ -1,950 +0,0 @@
-import {ComponentFixture, TemplateFixture, createComponent} from './render_util';
-
-describe('JS control flow', () => {
-  it('should work with if block', () => {
```
Do we have an existing test with *ngIf or *ngFor for all of these? Want to check that we don't lose coverage before deleting.

AndrewKushnir commented:
@cexbrayat @kara does it make sense to remove the JS blocks tests alongside with the code (core, compiler) that supports JS blocks?

cexbrayat commented:
@kara I removed the second commit and rebased. The PR now only contains deletions of tests in `di_spec.ts`, that I already migrated to `acceptance` with `*ngIf/*ngFor`.

kara reviewed (approved):
LGTM

kara commented:
merge-assistance: Saucelabs failure was due to master being red
